### PR TITLE
HW3.1

### PR DESCRIPTION
**Pull request by ivkarmazin**

### CODE REVIEW

File: HW3/hw3.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import requests
+import json
+
+app = Flask(__name__)
+def random_birthday():
+    start= datetime.strptime('1964-01-01', '%Y-%m-%d')
+    end = datetime.strptime('2006-12-31', '%Y-%m-%d')
+    return start + timedelta(days=random.randint(0, int((end - start).days)))
+
+@app.route('/generate_users', methods=['GET'])
```
Comment (mikolaz27) [inline HW3/hw3.py:21]: use webargs

File: HW3/hw3.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+def generate_students():
+    # count should be as input GET parameter
+    count = int(request.args.get('count', 10))
+    if not isinstance(count, int):
+        return jsonify({'error': 'Input is not int'}), 400
+
+    faker_instance = Faker("uk_UA")
+    df = pd.DataFrame()
+    # set limit as 1000
+    for i in range(min(count, 1000)):
```
Comment (mikolaz27) [inline HW3/hw3.py]: 1000 is not default value this is max value, if user use more than 1000 you need to notify

File: HW3/hw3.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        # first_name, last_name, email, password, birthday (18-60)
+        new_row = {
+            'name': faker_instance.first_name(),
+            'last_name': faker_instance.last_name(),
+            'email': faker_instance.email(),
+            'password': faker_instance.password(),
+            'birthday': random_birthday().strftime('%Y-%m-%d')
```
Comment (mikolaz27) [inline HW3/hw3.py:43]: age has limits, check requirements

File: HW3/hw3.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    <head><title>Fake Users</title></head>
+    <body>
+        <h1>Fake Users</h1>
+        {table_html}
+    </body>
+    </html>
+    """
+    return html
+
+@app.route('/bitcoin_rate', methods=['GET'])
```
Comment (mikolaz27) [inline HW3/hw3.py:68]: use webargs

File: HW3/hw3.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+def get_bitcoin_value():
+    # /bitcoin_rate?currency=UAH&convert=100
+    # input parameter currency code - default is USD
+    # input parameter count and multiply by currency (int) - default count is 1
+    currency = request.args.get('currency', 'USD')
+    count = request.args.get('count', 1)
+
+    # return value currency of bitcoin
```
Comment (mikolaz27) [inline HW3/hw3.py:77]: check response status code

File: HW3/hw3.py [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    url = "https://test.bitpay.com/rates/BTC/" + currency
+
+    headers = {
```
Comment (mikolaz27) [inline HW3/hw3.py]: this not required

File: HW3/hw3.py [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+        "accept": "application/json",
+        "Content-Type": "application/json",
+        "X-Accept-Version": "2.0.0"
+    }
+
+    response = requests.get(url, headers=headers)
+
+    return str(float(json.loads(response.text)["data"]["rate"]) \
```
Comment (mikolaz27) [inline HW3/hw3.py]: make some variables here, do not make it complicated

File: HW3/hw3.py (hunk 3 above)
Comment (ivkarmazin) [reply HW3/hw3.py:43]: see very first function, def random_birthday():

**Pull request merged**